File: BackEnd/backend/src/main/java/com/fitapp/backend/routinecomplete/routineexercise/aplication/dto/response/RoutineSetParameterResponse.java
package com.fitapp.backend.routinecomplete.routineexercise.aplication.dto.response;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@JsonInclude(JsonInclude.Include.NON_NULL)
public class RoutineSetParameterResponse {
    
    @JsonProperty("id")
    private Long id;
    
    @JsonProperty("setTemplateId")
    private Long setTemplateId;
    
    @JsonProperty("parameterId")
    private Long parameterId;
    
    @JsonProperty("parameterName")
    private String parameterName;
    
    @JsonProperty("parameterType")
    private String parameterType;
    
    @JsonProperty("unit")
    private String unit;
    
    @JsonProperty("numericValue")
    private Double numericValue;
    
    @JsonProperty("durationValue")
    private Long durationValue;
    
    @JsonProperty("integerValue")
    private Integer integerValue;

    @JsonProperty("repetitions")
    private Integer repetitions;
}

File: FrontEnd/app/src/main/java/com/fitapp/appfit/response/sets/response/RoutineSetParameterResponse.kt
package com.fitapp.appfit.response.sets.response

import com.google.gson.annotations.SerializedName

data class RoutineSetParameterResponse(
    @SerializedName("id") val id: Long,
    @SerializedName("setTemplateId") val setTemplateId: Long,
    @SerializedName("parameterId") val parameterId: Long,
    @SerializedName("parameterName") val parameterName: String?,
    @SerializedName("parameterType") val parameterType: String?,
    @SerializedName("unit") val unit: String?,
    @SerializedName("repetitions") val repetitions: Int?,
    @SerializedName("numericValue") val numericValue: Double?,
    @SerializedName("durationValue") val durationValue: Long?,
    @SerializedName("integerValue") val integerValue: Int?
)